_posts/2023-17-10-remote-attestation-blog-part1.markdown

```diff
@@ -0,0 +1,137 @@
+
+<h3 style="text-align left; color:#6B98BF;"> Measured Boot and Secure Boot Go Hand in Hand </h3>
+
+These two processes go hand-in-hand to ensure trusted Operating System (O/S) boots. Measured Boot guards the system from the processor powering on to the point where the operating system is ready to run. The issue is that once the O/S is booted, Measured Boot stops and its output is encapsulated into the measured boot log and the PCR values in the TPM. Now that the O/S is unguarded, it is **conceivable** that someone could manage to navigate to the boot event log, the PCRs, and the TPM and alter them retroactively. They could even go as far as to replace the TPM device with a virtual TPM. 
+
+Secure Boot depends on Measured Boot to guard the UEFI Bios. Measured Boot will check the competent and take a hash of it to be stored for later to help prove whether or not it has been tampered with. The UEFI Bios will prevent an unsigned kernel from booting properly if the hashes taken do not match at the time and will boot the kernel in “lockdown mode,” which will prevent alterations to it. This makes making retroactive alteration of the measured boot output harder. 
+
+Measured Boot guards the UEFI Bios during the secure boot process by taking a hash at its step in the boot process. Secure boot guards the post-boot integrity of the kernel by keeping it in “lockdown mode” in case something suspicious happens. Both are needed to get to a point where we have a completely booted system with a kernel that can be trusted. 
+
+<h3 style="text-align left; color:#6B98BF;"> Integrity Measurement Architecture (IMA) for Continuous Attestation </h3>
+
+The two previously described aspects of Remote Attestation are helpful for a one-time check to see if the server you have provisioned was altered before it was booted and running. But what can you do if you want to continuously make sure that things aren’t being altered in real-time? You can use Linux kernel's Integrity Measurement Architecture (IMA) for that. Implementing IMA with your Remote Attestation framework lets you detect if files have been accidentally or maliciously altered. You can, both remotely and locally, appraise a file's current measurement against a "good" value stored as an extended attribute and enforce local file integrity. IMA collects file hashes and places them in kernel memory where other applications cannot access or modify them. With Secure and Measured Boot enabled, we established a chain of trust from the TPM all the way up to the running kernel. Because the kernel is now trusted, we can trust it to measure files with IMA, thus extending that trust to the files. 
```

Comment:
In IMA appraisal normally means local (in the sense of on the device) file integrity enforcement. Maybe only introduce only the measurement part first, because the appraisal is explained in the next section in more detail.